src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -913,4 +914,45 @@ def parquet(
             **copy_options,
         )
 
+    @publicapi
+    def insert_into(
+        self, table_name: Union[str, Iterable[str]], overwrite: bool = False
+    ) -> None:
+        """
+        Inserts the content of the DataFrame to the specified table.
+        It requires that the schema of the DataFrame is the same as the schema of the table.
+
+        Args:
+            table_name: A string or list of strings representing table name.
+                If input is a string, it represents the table name; if input is of type iterable of strings,
+                it represents the fully-qualified object identifier (database name, schema name, and table name).
+            overwrite: If True, the content of table will be overwritten.
+                If False, the data will be appended to the table. Default is False.
+
+        Example::
+
+            >>> # save this dataframe to a json file on the session stage
+            >>> df = session.create_dataframe([["John", "Berry"]], schema = ["FIRST_NAME", "LAST_NAME"])
+            >>> df.write.save_as_table("my_table", table_type="temporary")
+            >>> df2 = session.create_dataframe([["Rick", "Berry"]], schema = ["FIRST_NAME", "LAST_NAME"])
+            >>> df2.write.insert_into("my_table")
+            >>> session.table("my_table").collect()
+            [Row(FIRST_NAME='John', LAST_NAME='Berry'), Row(FIRST_NAME='Rick', LAST_NAME='Berry')]
+        """
+        full_table_name = (
+            table_name if isinstance(table_name, str) else ".".join(table_name)
+        )
+        validate_object_name(full_table_name)
+        qualified_table_name = (
+            parse_table_name(table_name) if isinstance(table_name, str) else table_name
+        )
+
+        target_table = self._dataframe._session.table(qualified_table_name)
```

Comment:
does it work in pyspark even when table does not exist? I think this would fail if table doesn't exist.

Reply:
pyspark would fail.
The non-existing table error would be triggered by `if target_table.schema != self._dataframe.schema:` (this check serves both table check and schema check), I have a negative test for non-existing table